irods/session.py

```diff
@@ -141,7 +142,7 @@ def __init__(self, configure = True, auto_cleanup = True, **kwargs):
         self._env_file = ''
         self._auth_file = ''
         self.do_configure = (kwargs if configure else {})
-        self._cached_connection_timeout = kwargs.pop('connection_timeout', DEFAULT_CONNECTION_TIMEOUT)
+        self.connection_timeout = kwargs.pop('connection_timeout', DEFAULT_CONNECTION_TIMEOUT)
```

Comment:
Should we continue to set `_cached_connection_timeout` here? It looks like the `configure()` method is setting the connection pool timeout to `None` if `_cached_connection_timeout` is not set, and we call `configure` just a few lines below here in the default case.

If we should NOT be setting `_cached_connection_timeout` here, perhaps the reason should go in a comment here? Or, better yet, is there a reason to keep `_cached_connection_timeout` at all?

Reply:
The session attribute `_cached_connection_timeout` exists as a place to store the current session timeout value, whether or not a resident pool has been created. It's possible there is some redundancy here, I'll take a look at it.

Reply:
Went ahead with initializing `_cached_connection_timeout` in the session c'tor, because it makes sense to.

Reply:
Please resolve if this has been addressed. It seems to me that it has. Thanks